datajoint/base_relation.py

```diff
@@ -70,8 +71,14 @@ def declare(self):
         Loads the table heading. If the table is not declared, use self.definition to declare
         """
         if not self.is_declared:
-            self.connection.query(
-                declare(self.full_table_name, self.definition, self._context))
+            if self.definition is not None and self.definition is not Ellipsis:
+                self.connection.query(
+                    declare(self.full_table_name, self.definition, self._context))
+            else:
+                name = self.__class__.__name__
+                warn("""Table for %s is not declared and definition is not defined.
```

Comment:
Because that is what I want to avoid. Consider the following situation. 
* somebody writes a matlab schema and I'd like to write some functions for that tables in python
* therefore I define the corresponding classes in python
* now somebody drops some of the tables from matlab because (s)he wants to repopulate it
* so far I cannot even import my schema anymore until the tables are redefined in the database
* I am fine if my code fails if I want to instantiate an undeclared table, but I do want to be able to import my other objects to still use them until the dropped table is redfined in the database.